keps/sig-storage/625-csi-migration/kep.yaml

```diff
@@ -49,6 +49,11 @@ feature-gates:
     - kube-controller-manager
     - kubelet
     - kube-scheduler
+  - name: InTreePlugin{cloud-provider}Unregister
```

Comment:
I think this is where some confusion is coming because it's related to but not strictly for CSI migration. @ehashman also had the suggestion that this may be better suited as a config rather than a feature gate, so I think we should revisit outside of the csi migration context.

cc @jsafrane for thoughts

Reply:
I think things will get way more complicated if we change to other formats. To move this forward, I suggest we keep the flag until CSI migration is fully done and intree code is removed. Change it to a config file does not simplify it

Reply:
Or maybe I can remove this section and just remove the intreexxx feature gate from the vendor specific feature flags since technically they do not belong to this KEP

Reply:
I discussed with @jsafrane and we also think keeping it as a feature gate is simpler. Eventually once the feature gate is GA and intree code is removed, then the feature gates will also be removed.

With a config option, we would have to deal with deprecating the config API and following the deprecation timeline.

Reply:
To me, feature gates have a clear path towards GA and then they're removed. With a config (new field in component configs like [KubeletConfiguration](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/config/v1beta1/types.go#L818)?), this API will be more permanent and harder to remove when everything is migrated to CSI.

Reply:
Another challenge with the config is we would need to create new config APIs for many kubernetes components: kubelet, kube-controller-manager, kube-scheduler

Reply:
So I did some digging and it seems that these `InTreePlugin{Provider}Unregister` flags were added in the 1.21 cycle (https://github.com/kubernetes/website/blob/ed9728ca8cbf5f6933bfd3ad66eb22a3582937fb/content/en/docs/reference/command-line-tools-reference/feature-gates.md) but weren't documented in the KEP. So it probably doesn't make sense to revisit the design at this point, but I do think they need to be properly documented and discussed here.

The flag should be added to all applicable PRR questions, not just the first one.

Reply:
> The flag should be added to all applicable PRR questions, not just the first one.
This flag is only relevant with the first question. It is kind of standalone feature flag but happens to play a role in the overall CSI Migration feature. But technically it is not CSI migration specific feature gate and can be managed separately. So here I create a table to describe what happens with all the different scenario and hope if makes more sense.